### PR TITLE
Merge onboard, offboard and hybrid routers

### PR DESCRIPTION
**Pull request by LukasPaczos**
This PR consolidates all router modules into one. This removes the granular modularity of onboard and offboard router modules and leaves only one routing engine implementation that encapsulates all of the hybrid logic. This opens the door for Mapbox to improve this logic in the future by centralizing it in a native codebase available for all platforms which wouldn't be possible with onboard and offboard routers being modular as well.

The default, hybrid router is still modular by excluding its dependency:
```
implementation("com.mapbox.navigation:core:1.0.0") {
    exclude group: "com.mapbox.navigation", module: "router"
}
```
 and providing a custom implementation annotated with:
```
@MapboxModule(MapboxModuleType.NavigationRouter)
```

This PR also updates the routers package:
```
com.mapbox.navigation.route -> com.mapbox.navigation.router
```
and the module name:
```
libdirections-hybrid -> libnavigation-router
```

### CODE REVIEW

**Comment by LukasPaczos** [comment]
> This PR also updates the routers package

After looking at the diffs, this seems unnecessary and doesn't match the base interface anymore so I'm gonna go ahead and revert this change.

**Comment by RingerJK** [comment]
Do I get right, that from now we don't have separate offline and online routing? And if user want to replace offline router he doesn't have that option?

**Comment by LukasPaczos** [comment]
That's right @RingerJK. Users will still be able to replace the routing logic as a whole, but not only the offline routing or only the online routing. Ultimately, we haven't heard any specific request for this much control, so we might always reconsider in the future if those come.

**Comment by RingerJK** [comment]
@LukasPaczos get it. From this view perspective we should reinvestigate `RouteUrl` class. One was created because Offline navigation(`onboard` module) cannot consume network-dependency libraries, but seems that now it might 😄

File: libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/hybrid/MapboxHybridRouter.kt [hunk 1]
```diff
@@ -28,12 +34,38 @@ class MapboxHybridRouter(
     networkStatusService: NetworkStatusService
 ) : Router {
 
+    constructor(
```
Comment (RingerJK) [inline libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/hybrid/MapboxHybridRouter.kt:37]: never used, may be avoid create additional constructor 🤔

Comment (RingerJK) [reply libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/hybrid/MapboxHybridRouter.kt:37]: Oh, I got it. resolved

Comment (LukasPaczos) [reply libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/hybrid/MapboxHybridRouter.kt:37]: This constructor is used when injecting the module in the `MapboxNavigation`.

**Comment by LukasPaczos** [comment]
> One was created because Offline navigation(onboard module) cannot consume network-dependency libraries, but seems that now it might

Yup, good point :+1: I'm going to ticket that as tail work.

**Review by RingerJK — APPROVED** [review]

**Pull request merged**